beacon-chain/forkchoice/protoarray/helpers_test.go

```diff
@@ -0,0 +1,325 @@
+package protoarray
+
+import (
+	"context"
+	"encoding/binary"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+func TestComputeDelta_ZeroHash(t *testing.T) {
+	validatorCount := uint64(16)
+	indices := make(map[[32]byte]uint64)
+	votes := make([]Vote, 0)
+	oldBalances := make([]uint64, 0)
+	newBalances := make([]uint64, 0)
+
+	for i := uint64(0); i < validatorCount; i++ {
+		indices[indexToHash(i)] = i
+		votes = append(votes, Vote{params.BeaconConfig().ZeroHash, params.BeaconConfig().ZeroHash, 0})
+		oldBalances = append(oldBalances, 0)
+		newBalances = append(newBalances, 0)
+	}
+
+	delta, _, err := computeDeltas(context.Background(), indices, votes, oldBalances, newBalances)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(delta) != int(validatorCount) {
+		t.Error("Incorrect length")
+	}
+	for _, d := range delta {
+		if d != 0 {
+			t.Error("Delta should be zero")
+		}
+	}
+	for _, vote := range votes {
+		if vote.currentRoot != vote.nextRoot {
+			t.Errorf("The vote should have changed")
```

Comment:
Should this error say "The vote should not have changed"? Since the condition above it is true if the did change?

Reply:
"Should have changed" is correct. Current vote and next vote are different to begin with. After node's processed next vote, it sets next vote to current vote and the next next vote will be different 👍🏼